utils/dsl/validation.ts

```diff
@@ -0,0 +1,108 @@
+import { DSL, Expression } from "../../models/dsl";
+
+const validateSide = (
+  param: string | number | Expression,
+  side: string,
+): [boolean, string] => {
+  // 1. Validate field type
+  if (
+    typeof param !== "string" &&
+    typeof param !== "number" &&
+    typeof param !== "object"
+  ) {
+    return [false, `${side} field is not a valid type: ${param}`];
+  }
+
+  // 2. If side is an object, recursively validate the expression inside of it again.
+  if (typeof param === "object") {
+    const [success, error] = validExpression(param);
+
+    if (!success) {
+      return [false, error];
+    }
+  }
+
+  return [true, ""];
+};
+
+const validExpression = (expression: Expression): [boolean, string] => {
+  // 1. Check that the expression contains the relevant fields we expect AND only those fields.
+  if (!expression.fn || !expression.a || !expression.b) {
```

Comment:
This check doesn't allow a or b to be 0 due to it being falsey. 

If 0 was a valid value for a or b, then this would be the condition:
`if (
    !expression.fn ||
    (!expression.a && expression.a !== 0) ||
    (!expression.b && expression.b !== 0)
  )`